spring-cloud-gcp-samples/spring-cloud-gcp-storage-resource-sample/src/main/java/com/example/WebController.java

```diff
@@ -35,17 +39,45 @@
 @RestController
 public class WebController {
 
+  @Value("${gcs-resource-test-bucket}")
+  private String bucketName;
+
   @Value("gs://${gcs-resource-test-bucket}/my-file.txt")
   private Resource gcsFile;
 
-  @GetMapping("/")
-  public String readGcsFile() throws IOException {
+  private Storage storage;
+
+  private WebController(Storage storage) {
+    this.storage = storage;
+  }
+
+  @GetMapping(value = "/")
+  public String readGcsFile(@RequestParam("filename") Optional<String> fileName)
+      throws IOException {
+    if (fileName.isPresent()) {
+      GoogleStorageResource resource =
+          new GoogleStorageResource(
+              this.storage, String.format("gs://%s/%s", bucketName, fileName.get()));
```

Comment:
Consider a `private Resource getResource(Optional<String> filename)` function to reduce the duplication in this method and `writeGcs`.

```java
  @GetMapping(value = "/")
  public String readGcsFile(@RequestParam("filename") Optional<String> fileName)
      throws IOException {
    Resource resource = getResource(fileName);
    return StreamUtils.copyToString(resource.getInputStream(), Charset.defaultCharset()) + "\n";
  }
```